src/mlpack/tests/recurrent_network_test.cpp

```diff
@@ -1413,4 +1413,68 @@ BOOST_AUTO_TEST_CASE(BRNNTrainReturnObjective)
   BOOST_REQUIRE_EQUAL(std::isfinite(objVal), true);
 }
 
+/**
+ * Test that RNN::Train() does not give an error for large rho.
+ */
+BOOST_AUTO_TEST_CASE(LargeRhoValueRnnTest)
+{
+  // Setting rho value greater than sequence length which is 17
+  const int rho = 100;
+  const size_t hiddenSize = 128;
+  const size_t numLetters = 256;
+  using namespace arma;
+  using MatType = cube;
+  std::vector<std::string> trainingData;
+  trainingData.push_back(std::string("THIS IS THE INPUT"));
+
+  RNN<> model(rho);
+  model.Add<IdentityLayer<>>();
+  model.Add<LSTM<>>(numLetters, hiddenSize, rho);
+  model.Add<Dropout<>>(0.1);
+  model.Add<Linear<>>(hiddenSize, numLetters);
+
+  const auto makeInput = [numLetters](const char *line) -> MatType
+  {
+    const auto strLen = strlen(line);
+    // rows: number of dimensions
+    // cols: number of sequences/points
+    // slices: number of steps in sequences
+    MatType result(numLetters, 1, strLen, fill::zeros);
+    for (size_t i = 0; i < strLen; ++i)
+    {
+      const auto letter = line[i];
+      result.at(static_cast<uword>(letter), 0, i) = 1.0;
+    }
+    return result;
+  };
+
+  const auto makeTarget = [] (const char *line) -> MatType
+  {
+    const auto strLen = strlen(line);
+    // Responses for NegativeLogLikelihood should be
+    // non-one-hot-encoded class IDs (from 1 to num_classes).
+    cube result(1, 1, strLen, fill::zeros);
```

Comment:
If you already used a typedef to set what `MatType` is, shouldn't you use that here too?  (and in other places in this method)